### PR TITLE
Feature: POST user endpoint

### PR DESCRIPTION
**Pull request by Miles-Alford**
### *What is the problem we're trying to solve*

This PR adds the first version of the POST user endpoint, this endpoint will allow users to be added to the system to allow future functionality such as allowing a user to view the projects they are assigned too.

### *What changes have we introduced*
Quite a lot of changes made such as deletion of most of the unused example files from the base api template, creation of the complete endpoint from controller to gateway, the addition of the first migration to allow the creation of a user table, updates to the `StartUp` class and changes made to allow docker to run the migrations within the container databases

### *Follow up actions after merging PR*
Some improvements still need to be made to the endpoint, mainly in terms of validation such as checking if the user already exists, this will be done once the GET endpoint has been created

### CODE REVIEW

File: ProjectFinderApi/Startup.cs [hunk 1]
```diff
@@ -122,7 +126,7 @@ public void ConfigureServices(IServiceCollection services)
 
         private static void ConfigureDbContext(IServiceCollection services)
         {
-            var connectionString = Environment.GetEnvironmentVariable("CONNECTION_STRING");
+            var connectionString = Environment.GetEnvironmentVariable("CONNECTION_STRING") ?? "Host=;Database=;";
```
Comment (LBHTKarki) [inline ProjectFinderApi/Startup.cs:129]: This looks like it's missing some values

**Review by LBHTKarki — CHANGES_REQUESTED** [review]
Couple of things missing, but overall looks very good!

File: ProjectFinderApi.Tests/V1/Gateways/UsersGatewayTests.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+using NUnit.Framework;
+using ProjectFinderApi.Tests.V1.Helpers;
+using ProjectFinderApi.V1.Gateways;
+
+namespace ProjectFinderApi.Tests.V1.Gateways
+{
+    public class UsersGatewayTests : DatabaseTests
+    {
+        private UsersGateway _classUnderTest;
+        private Faker _faker;
```
[the remark this answers is not in the document]
Comment (Miles-Alford) [reply ProjectFinderApi.Tests/V1/Gateways/UsersGatewayTests.cs]: Good call, don't even need them in this test anyway, they are left overs from an idea I had earlier

File: ProjectFinderApi/Startup.cs (hunk 1 above)
Comment (Miles-Alford) [reply ProjectFinderApi/Startup.cs:129]: This is just a weird way to get around running a migration locally, I tried using profiles but it wouldn't use the connection string within it and found this line within another API, I think it doesn't care what string you use as running a migration only requires that a connection string has been set

File: ProjectFinderApi/V1/Controllers/UserController.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            {
+                return BadRequest(validationResults.ToString());
+            }
+
+            try
+            {
+                var createdUser = _usersUseCase.ExecutePost(request);
+                return CreatedAtAction("User successfully created", createdUser);
+            }
+            catch (PostUserException e)
```
[the remark this answers is not in the document]
Comment (Miles-Alford) [reply ProjectFinderApi/V1/Controllers/UserController.cs:45]: I'm thinking of throwing this exception once I have in place the validation check of if a user already exists within the database

**Review by LBHTKarki — APPROVED** [review]
I have looked at your comments regarding what's planned next, so it all looks good. Good practise would be to remove all code that's not currently relevant or relies on future features. That can be easily addressed by smaller PRs going forward. Great start!

**Pull request merged**